### DISCUSSION TITLE
Differend websockets states in endpoint and exception handler

### DISCUSSION THREAD

**Discussion by Whatislove118** [fastapi/fastapi#6043] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.websocket("/ws/")
async def execute_command(
    socket: CommunicationSocket = Depends(get_communication_socket),
) -> None:
    manager = socket.get_manager()
    await manager.connect(socket.websocket)
    logger.info(f"Client {socket.app_resolver.app.name} connected")
    try:
        await manager.process(socket)
    except WebSocketDisconnect:
        manager.disconnect(socket.websocket)
        logger.info(f"Client {socket.app_resolver.app.name} disconnected")
    except pydantic.ValidationError as e:
        raise CommunicationValidationError(str(e))


## EXCEPTION HANDLER

async def handle_service_exception(request: HTTPConnection, exc: LimIDException) -> JSONResponse:
    exception_data = exc.data
    status_code = exception_data.status_code[request.scope["type"]]

    if status_code is None:
        raise exc

    if isinstance(request, WebSocket):
        # request.application_state = WebSocketState.CONNECTED # this allows it to work only if the socket is open (i.e. one has to try to guess)
        if request.application_state == WebSocketState.CONNECTING:
            await request.accept()
            
        
        await request.send_json({"message": exception_data.message, "status": False, "code": status_code})
        await request.close(code=int(status_code), reason=exception_data.message)

    return JSONResponse({"message": exception_data.message, "status": False}, status_code=status_code)
```

### Description

In endpoint, I open and accept web socket connection. State of web socket set to CONNECTED.
If an exception raised in endpoint, the exception handler should process that. But request web socket doesn't equal to endpoint web socket by state (always CONNECTING in handler).

On connect:
![image](https://user-images.githubusercontent.com/56884186/207064790-2b2f7011-787e-40ec-abd4-4182b3b9e11c.png)

After exception raised:
![image](https://user-images.githubusercontent.com/56884186/207064993-503afc1d-8c3b-433c-be8a-4802c1cb6208.png)

And finally, send and receive are common, I can send/receive message from this "different" web sockets

### Operating System

macOS

### FastAPI Version

0.88.0

### Python Version

3.11

**Comment by iudeen** [comment]
Is this your complete code (MRE)? Where are you attaching your custom exception handler?

**Comment by Whatislove118** [comment]
> Is this your complete code (MRE)? Where are you attaching your custom exception handler?

On startup to app (FastAPI instance)
![image](https://user-images.githubusercontent.com/56884186/207119429-159fb056-9864-4847-95ee-d22bc0139916.png)
